pkgs/os-specific/darwin/apple-source-releases/xnu/default.nix

```diff
@@ -150,7 +154,9 @@ appleDerivation' (if headersOnly then stdenvNoCC else stdenv) (
     mv $out/Library/Frameworks/IOKit.framework $out/Library/PrivateFrameworks
   '';
 
-  appleHeaders = builtins.readFile (./. + "/headers-${arch}.txt");
+  appleHeaders = let
+    name = if stdenv.hostPlatform.darwinSdkVersion == "10.12" then "headers-10.12-${arch}.txt" else "headers-${arch}.txt";
```

Comment:
Couldn’t this just be `headers-10.12.txt` and `headers-11.0.txt`?

Reply:
it can. i thought that the naming would cause rebuilds but it doesn't so i can align everything to use headers-${sdk-version}-${arch}.txt